Tags

nadiai opened the pull request:
You don't have to review the css file!

nadiai committed:
06/14 First commit

More images so i can make articles and tags will make sense

nadiai committed:
6/15 Commit

I completed tags! Now its time to redesign!

ohkhaila commented on Blog/include/functions.php:
```diff
@@ -54,6 +54,26 @@ function printHeader(){
+		WHERE tagID = :tagID
+		",
+	 array('tagID' => $tagID )
+ );
```
style suggestion: I would align this line 73 with line 67. 
Wherever something is opened it should close in the same relative position

ohkhaila reviewed (commented):
This looks great. My main piece of advice is to work on styling your code. (Good variable/function names that tell exactly what is done, aligning opening and closing statements, etc). Tyler will review this pull request as well.

tilikang commented on Blog/include/functions.php:
```diff
@@ -54,6 +54,26 @@ function printHeader(){
+}
+
+function getTagBlogs($tagID){
+	$result = dbQuery("
```
I can't see your schema, but i'm guessing that `PostTag` is a table with just a PostId and TagId? In that case, all this would return is some ids, not the actual info you want, right? Normally you'd want to join this with another table.

nadiai committed:
06/19 First Commit!

I fixed a bug now it works correctly! I made a second nav bar that i need to figure out how to correctly link but for now I'll leave it be!

tilikang commented on Blog/include/functions.php:
```diff
@@ -54,6 +54,27 @@ function printHeader(){
+function getTagBlogs($tagID){
+	$result = dbQuery("
+		SELECT *
+		FROM PostTag
```
It looks to me like this is only retrieving the PostTag table which I assume only has a `postId` and `tagId` field. It doesn't contain info on the actual tags or the actual posts, right?

If you join this query with the post table, you'd be able to see all the info about the posts.

Side note: for some reason you're not supposed to use capital letters in table names. I'm not sure why, but it's probably a good idea to rename your tables so they're all lowercase (for mysql tables I use the `post_tag` format)

tilikang commented on Blog/include/header.php:
```diff
@@ -9,40 +9,85 @@
+							<br/>
+						<?php
 
+							$posts = GetAllBlogPosts();
```
Something to consider for the future: if you end up with more and more blog posts, you might want to have this query limit the number returned so it only gets e.g. the last 10 posts.

tilikang commented on Blog/view/blogposts.php:
```diff
@@ -1,10 +1,13 @@
-<html>
+<?php
+include('Blog/include/functions.php');
+$print = printHeader();
```
You're saving this value to `$print` but from what I can tell, you aren't doing anything with `$print` later on in the code. Is this serving a purpose?

tilikang commented on Blog/view/sessions.php:
```diff
@@ -0,0 +1,11 @@
+
+session_start();
+
+if(!isset($_SESSION['userName'])){
```
If this is just practice code, that's fine. If it's the real login system, I'd suggest storing the userID (i.e. the primary key of the user table). There are a variety of reasons for this, but the main one is that internally, the primary key is meant to be the "pointer" to a row in your db. The username is just the thing a user can remember easily, but it's not ideal to use as a pointer in your internal code.

tilikang commented on Blog/view/tags.php:
```diff
@@ -0,0 +1,48 @@
+
+		<?php
+		 if ($_REQUEST['tagID'] == 0) {
+			echo "<title class='tagTitle'> Life Updates</title>";
```
As a general rule, you should try to avoid situation where your code directly references primary keys in your db (e.g. a specific `tagID`).

Instead, take the tagID out of the request and call the database to get that tag's info. Then use that info to display the correct name in this title tag.

tilikang reviewed (commented):
@nadiai back to you! I'd recommend merging both your PRs in as soon as you've addressed the comments.

CC @ohkhaila

nadiai committed:
06/21 First commit

Spent all morning working on one comment so I'm mad and moving on

nadiai committed:
06/21 second commit

nadiai committed:
06/21 Final Commit

Resolving PR comments

nadiai committed:
Merge branch 'staging' into tags
